lib/geminabox/gem_list_merge.rb

```diff
@@ -1,23 +1,38 @@
+require "set"
+
 module Geminabox
   class GemListMerge
     attr_accessor :list
 
     IGNORE_DEPENDENCIES = 0..-2
 
-    def self.from(*lists)
-      lists.map{|list| new(list)}.inject(:merge)
+    def self.merge(strategy, lists)
+      lists.map{|list| new(list)}.inject{|a,b| a.merge(b, strategy) }
     end
 
     def initialize(list)
       @list = list
     end
 
-    def merge(other)
+    def merge(other, strategy = nil)
+      send(strategy || :ours, other)
+    end
```

Comment:
We could avoid metaprogramming here and improve performance by skipping a `method_missing` lookup by using a `case/when` on `strategy` and returning a new custom exception (`UnsupportedMergeStrategy` perhaps?) We really don't have a plethora of strategies that we need to support.

Reply:
Yep! Switched over to simple strategy objects responding to `#merge`.